core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -87,7 +85,7 @@
                String operation,
                Map<String, String> summary,
                List<ManifestFile> dataManifests) {
-    this(io, INITIAL_SEQUENCE_NUMBER, snapshotId, parentId, timestampMillis, operation, summary, null);
+    this(io, TableMetadata.INITIAL_SEQUENCE_NUMBER, snapshotId, parentId, timestampMillis, operation, summary, null);
```

Comment:
This change looks unrelated to me. Can you remove it?

Reply:
Trying to piggy-back a change to consolidate `INITIAL_SEQUENCE_NUMBER` from multiple places. I can put it in another PR if it is preferred.

Reply:
Yes, please do. We try to avoid unnecessary changes so that we don't cause unnecessary commit conflicts for people maintaining branches.

Reply:
Removed it in a new commit.